IntegrationWithPharmacies/RabbitMQService.cs

```diff
@@ -0,0 +1,78 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using HealthClinic.CL.DbContextModel;
+using HealthClinic.CL.Dtos;
+using HealthClinic.CL.Model.ActionsAndBenefits;
+using HealthClinic.CL.Service;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Converters;
+using RabbitMQ.Client;
+using RabbitMQ.Client.Events;
+
+namespace IntegrationWithPharmacies
+{
+    public class RabbitMQService : BackgroundService
+    {
+        IConnection connection;
+        IModel channel;
+        public override Task StartAsync(CancellationToken cancellationToken)
+        {
+            var factory = new ConnectionFactory() { HostName = "localhost" };
+            connection = factory.CreateConnection();
+            channel = connection.CreateModel();
+            channel.QueueDeclare(queue: "hello",
+                                    durable: false,
+                                    exclusive: false,
+                                    autoDelete: false,
+                                    arguments: null);
+
+            var consumer = new EventingBasicConsumer(channel);
+            consumer.Received += (model, ea) =>
+            {
+                byte[] body = ea.Body.ToArray();
+                var jsonMessage = Encoding.UTF8.GetString(body);
+                Message message;
+                try
+                {   
+                    message = JsonConvert.DeserializeObject<Message>(jsonMessage);
+                }
+                catch (Exception) 
+                {
+                    message = JsonConvert.DeserializeObject<Message>(jsonMessage, new MyDateTimeConverter());
+                }
+
+                message.PharmacyName = "Apoteka Jankovic";
+                Program.ListOfMessages.Add(message);
+
+                MessageService messageService = new MessageService();
+                MessageDTO dto = new MessageDTO(message.Text, message.Timestamp, message.PharmacyName);
+                messageService.Create(dto);

```

Comment:
The value of the variable 'dto' can be passed directly as a parameter to the 'Create ()' method.